### ISSUE TITLE
Doesn't switch device when changed

### ISSUE THREAD

**Issue by confact** [jpochyla/psst#215]
**Describe the bug**
I am using latest commit (96059d95d6a69294c0847bfd0f67d0f6dda11e47) and it works good on my mac. The only thing is that I can start playing and then plug in a Bluetooth headset, and it will continue to play in the speakers while everything else goes through the headset.

**To Reproduce**
* start playing music
* trigger device change (plugin headset, Bluetooth, etc.)
* It will continue playing on the previous device

**Expected behavior**
I want it to change to the new default device - listening on sound device changes.

**Screenshots**

**Environment**
 - OS: MacOS 
 - Version: 10.15.7 (19H1323)

**Additional context**

**Comment by jpochyla** [comment]
Yup, this is unfortunately a limitation of CPAL at the moment https://github.com/RustAudio/cpal/issues/373

**Comment by confact** [comment]
@jpochyla, okay. I would love it to be fixed, as I have to restart Psst 3-4 times a day now every time I connect my BT headset.
I will follow that issue for sure! Do you have any ideas what I could do until then?

**Comment by jpochyla** [comment]
It annoys me as well, I'll look into it.

**Comment by SO9010** [comment]
@jacksongoode is this an issue for you? I can test on windows soon to see if its still an issue there. On linux its fine.
